astropy/table/operations.py

```diff
@@ -697,6 +697,69 @@ def common_dtype(cols):
         raise tme
 
 
+def _get_join_sort_idxs(out_descrs, keys, len_left, len_right, left, right):
+
+    # Remake out_descrs as a dict keyed by the name (descr[0])
+    descrs = {descr[0]: descr for descr in out_descrs}
+
+    # Go through each of the key columns in order and make columns for
+    # a new structured array that represents the lexical ordering of those
+    # key columns. This structured array is then argsort'ed. The trick here
+    # is that some columns (e.g. Time) may need to be expanded into multiple
+    # columns for ordering here.
```

Comment:
Thank you for these detailed comments, it really helps a lot now and will in the  future, too